src/transformers/tokenization_utils_base.py

```diff
@@ -1523,6 +1517,9 @@ def __init__(self, **kwargs):
         self.name_or_path = kwargs.pop("name_or_path", "")
         self._processor_class = kwargs.pop("processor_class", None)
 
+        if not hasattr(self, "_added_tokens_decoder"):
+            self._added_tokens_decoder: Dict[int, AddedToken] = {}
```

Comment:
If you have strange offsets, or if you don't have a unk token in the vocab, this is how you add them.:
```python 
self._added_tokens_decoder = {
0:unk_token,
4930:bos_token
}
```